rng_pre120.go

```diff
@@ -0,0 +1,15 @@
+//go:build !go1.20
+// +build !go1.20
+
+package rehttp
+
+import (
+	"math/rand"
+	"time"
+)
+
+// Only seed the global random source on versions prior to 1.20 since
+// it's pre-seeded starting in 1.20
+func init() {
+	rand.Seed(time.Now().UnixNano())
```

Comment:
I don't think the package should do this - the caller might have seeded the global source with a known value to get deterministic results. I think we should document on `ExpJitterDelay` that it assumes the global source has been properly seeded by the caller.